src/AWS/CRT/Internal/FFI.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace AWS\CRT\Internal;
+
+use \Exception;
+use \RuntimeException;
+
+/**
+ * @internal
+ * Forwards calls on to libaws-crt-ffi via FFI
+ */
+final class FFI {
+    private static $ffi = null;
+    private static $refcount = 0;
+
+    function __construct() {
+        if (is_null(self::$ffi)) {
+            try {
+                self::$ffi = \FFI::cdef(file_get_contents(__DIR__ . "/../../../api.h"), __DIR__ . "/../../../libaws-crt-ffi.so");
+                self::init();
+            } catch (Exception $e) {
+                throw new RuntimeException('Exception while initializing CRT via FFI', 0, $e);
+            }
+        }
+        ++self::$refcount;
```

Comment:
You are using postfix ++ in other places where the prefix vs postfix makes no difference too. Should we keep them all the same for consistency?

Reply:
Yeah, I was working quickly, I'll clean it up.